opentracing/noop.go

```diff
@@ -0,0 +1,62 @@
+// Noop implementations of the core opentracing interfaces.
+package opentracing
+
+import "golang.org/x/net/context"
+
+// noopContextID:
+type noopContextID struct{}
+
+func (n noopContextID) NewChild() (ContextID, Tags) {
+	return noopContextID{}, Tags{}
+}
+func (n noopContextID) Serialize() []byte {
+	return []byte{}
+}
+
+// noopSpan:
+type noopSpan struct{}
+
+func (n noopSpan) StartChildSpan(operationName string, parent ...context.Context) (Span, context.Context) {
+	if len(parent) > 0 {
+		return noopSpan{}, parent[0]
+	} else {
+		return noopSpan{}, context.Background()
+	}
+}
+func (n noopSpan) SetTag(key string, value interface{})           {}
+func (n noopSpan) Info(message string, payload ...interface{})    {}
+func (n noopSpan) Warning(message string, payload ...interface{}) {}
```

Comment:
I officially do not feel strongly about this and am happy to do whatever others think is most reasonable (as long as there's a way to distinguish errors from non-errors). I can see arguments either way for various levels of fidelity around log severity.